src/core/python2js_buffer.c

```diff
@@ -335,33 +335,34 @@ _python2js_buffer_to_typed_array(Py_buffer* buff)
     }
   }
 
+  Py_ssize_t len = buff->len / buff->itemsize;
   switch (format) {
     case 'c':
     case 'b':
-      return hiwire_int8array((i8*)buff->buf, buff->len);
+      return hiwire_int8array((i8*)buff->buf, len);
```

Comment:
Shouldn't we have been able to detect this in test, via the array length and if so could we add  a regression test.

Reply:
No, it's actually pretty weird, it first makes a too-big array in `_python2js_buffer_to_typed_array`:
https://github.com/iodide-project/pyodide/blob/69e28cbe0ef9e0292235575df1bc23a96cbb7815/src/core/python2js_buffer.c#L473
and then it slices the result out of it in `_python2js_shareable_buffer_recursive` 
https://github.com/iodide-project/pyodide/blob/69e28cbe0ef9e0292235575df1bc23a96cbb7815/src/core/python2js_buffer.c#L482
The result it makes has the right size. Then it leaks the too-big array. Only reason I noticed this internal miscalculation is that it leaked the object.
I think I should add a regression test for the memory leak though.

Reply:
OK, thanks for the clarification!